src/components/Messaging/AnnounceMessage.js

```diff
@@ -109,8 +115,41 @@ const AnnounceMessageForm = ({ setToggleAnnounceMessage }) => {
         (payload.body && payload.organization_id) ||
         (payload.body && payload.region_id)
       ) {
-        await postBulkMessageSend(payload);
-        history.go(0);
+        const res = await postBulkMessageSend(payload);
+
+        if (res.error) {
+          setErrorMessage(res.message);
+        } else {
+          const newAnnouncement = {
+            id: null,
+            type: 'announce',
+            parent_message_id: null,
+            from: payload.author_handle,
+            to: null,
+            recipient_organization_id: payload.organization_id || null,
+            recipient_region_id: payload.region_id || null,
+            subject: payload.subject,
+            body: payload.body,
+            composed_at: new Date().toISOString(),
+            video_link: null,
+            survey_response: null,
+            survey: null,
+            bulk_message_recipients: [],
+          };
+          log.debug('...update threads w/ new announcement');
+          // update the full set of threads
+          setThreads((prev) => {
+            const updated = [
+              {
+                username: `${newAnnouncement.id}`,
+                messages: [newAnnouncement],
+                avatar: '',
+              },
+              ...prev,
+            ];
+            return updated;
```

Comment:
Maybe you can simplify this and return directly the array instead of creating a variable and returning it.